test/e2e/data/kustomize/upgrade-patches/ci-hack-kcp.yaml

```diff
@@ -0,0 +1,31 @@
+---
+# Workaround for https://github.com/kubernetes-sigs/cluster-api/issues/7457
+# There is a small but important difference between these two:
+# path: /a/b/c
+# *creates* the c array, overwriting anything that was there before
+# path: /a/b/c/-
+# *adds* to the c array and does not work if the array is missing
+#
+# We add to the postKubeadmCommands (instead of pre*) since we need the CI artifacts
+# script to run first. Without this, the container images are not imported properly.
+- op: add
+  path: /spec/kubeadmConfigSpec/postKubeadmCommands
+  value:
+  - /usr/local/bin/ci-artifacts-openstack.sh
```

Comment:
Where does this run? Is it on the test runner or on the target? I assume the latter.